components/camel-minio/src/main/java/org/apache/camel/component/minio/client/GetMinioClient.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.minio.client;
+
+import io.minio.MinioClient;
+import org.apache.camel.component.minio.MinioConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Creates MinIO client object according to the
+ * given endpoint, port, access key, secret key, region and secure option.
+ */
+public class GetMinioClient implements MinioCamelInternalClient {
+    private static final Logger LOG = LoggerFactory.getLogger(GetMinioClient.class);
+    private final MinioConfiguration configuration;
+
+    /**
+     * Constructor that uses the config file.
+     */
+    public GetMinioClient(MinioConfiguration configuration) {
+        LOG.trace("Creating an Minio client.");
+        this.configuration = configuration;
+    }
+
+    /**
+     * Getting the minio client.
+     *
+     * @return Minio Client.
+     */
+    @Override
+    public MinioClient getMinioClient() {
+        if (configuration.getEndpoint() != null) {
+            MinioClient.Builder minioClientRequest = MinioClient.builder();
+
+            if (configuration.getProxyPort() != null) {
+                minioClientRequest.endpoint(configuration.getEndpoint(), configuration.getProxyPort(), configuration.isSecure());
+            } else {
+                minioClientRequest.endpoint(configuration.getEndpoint());
+            }
+            if (configuration.getAccessKey() != null && configuration.getSecretKey() != null) {
```

Comment:
I wonder accessKeys, same as region are not required?

Reply:
Hi @omarsmak sorry, I didn't get this question?

Reply:
What I mean here, are the `accessKey`, `secretKey` and `region` are required in order to initiate the client? What happens if one of these is null or not set?

Reply:
@omarsmak, As per [Java Client API Reference](https://docs.min.io/docs/java-client-api-reference.html) you can create MinioClient with or without ```accessKey```, ```secretKey``` (namely **anonymous access**) or ```region```.

Reply:
@omarsmak is it okay to mark this as resolved?